### PR TITLE
[SYCL][Matrix][E2E] make joint_matrix_transposeC_impl.hpp simpler: just joint_matrix_load&joint_matrix_store for colC

### PR DESCRIPTION
**Pull request by yubingex007-a11y**

### CODE REVIEW

**Comment by yubingex007-a11y** [comment]
I think we can merge this PR after my backend patch is merged

**Review by dkhaldi — APPROVED** [review]
LGTM

**Review by YuriPlyakhin — CHANGES_REQUESTED** [review]

**Comment by YuriPlyakhin** [comment]
Also should XFAIL be updated here as well? SG32/joint_matrix_transposeC.cpp

**Comment by yubingex007-a11y** [comment]
ping @YuriPlyakhin

File: sycl/test-e2e/Matrix/joint_matrix_transposeC_impl.hpp
Comment (yubingex007-a11y) [reply sycl/test-e2e/Matrix/joint_matrix_transposeC_impl.hpp]: done

File: sycl/test-e2e/Matrix/joint_matrix_transposeC_impl.hpp
Comment (yubingex007-a11y) [reply sycl/test-e2e/Matrix/joint_matrix_transposeC_impl.hpp]: done

File: sycl/test-e2e/Matrix/joint_matrix_transposeC_impl.hpp
Comment (yubingex007-a11y) [reply sycl/test-e2e/Matrix/joint_matrix_transposeC_impl.hpp]: done

**Comment by yubingex007-a11y** [comment]
> Also should XFAIL be updated in SG32/joint_matrix_transposeC.cpp as well?

done

**Review by YuriPlyakhin — APPROVED** [review]
LGTM

**Comment by YuriPlyakhin** [comment]
@intel/llvm-gatekeepers , all green, please, merge

**Pull request merged**
